presto-main/src/main/java/com/facebook/presto/operator/aggregation/DoubleSumAggregation.java

```diff
@@ -25,13 +25,29 @@
     private DoubleSumAggregation() {}
 
     @InputFunction
-    @IntermediateInputFunction
     public static void sum(NullableDoubleState state, @SqlType(StandardTypes.DOUBLE) double value)
     {
         state.setNull(false);
         state.setDouble(state.getDouble() + value);
     }
 
+    @CombineFunction
+    public static void combine(NullableDoubleState state, NullableDoubleState otherState)
+    {
+        if (state.isNull()) {
+            if (otherState.isNull()) {
+                return;
+            }
+            state.setNull(false);
```

Comment:
TL;DR `otherState.isNull()` is guaranteed to be false. Therefore you can remove `if`s that is guaranteed to go one way or the other.

There are 2 guarantees provided by the aggregation function framework:
* `otherState` is guaranteed to be the result of `deserialize`.
* When deserializing, `null` values are skipped.

For most states, including `NullableDoubleState`, null states are serialized as `null` value. Therefore, `otherState` cannot be a null state.

I understand that the guarantees provided here are complex. When the aggregation function framework came out, we didn't document them, in the hope that implementations will just do the right thing as if no such guarantee is provided. We ended up failing miserably, having inconsistent and self-contradictory code everywhere for aggregation functions. Yet these seemingly unreasonable code are all correct. (The tests that get run on aggregation functions covers null related corner cases pretty well.)

Up until recently, these guarantees are not documented.